types/setting.go

```diff
@@ -105,6 +107,7 @@ type SettingDefinition struct {
 	Required    bool            `json:"required"`
 	ReadOnly    bool            `json:"readOnly"`
 	Default     string          `json:"default"`
+	Options     []string        `json:"options,omitempty"` // +optional
```

Comment:
I prefer `Choices` instead of `Options`. Options sounds like command line parameters in the CS context.